src/cursor/cur_version.c

```diff
@@ -0,0 +1,546 @@
+                version_cursor->upd_durable_stop_ts = upd->durable_ts;
+                version_cursor->upd_stop_ts = upd->start_ts;
+
+                upd_found = true;
+                version_cursor->next_upd = upd->next;
+            }
+        }
+    }
+
+    if (!upd_found && !F_ISSET(version_cursor, WT_VERSION_CUR_ON_DISK_EXHAUSTED)) {
+        switch (page->type) {
+        case WT_PAGE_ROW_LEAF:
+            if (cbt->ins != NULL) {
+                F_SET(version_cursor, WT_VERSION_CUR_ON_DISK_EXHAUSTED);
+                F_SET(version_cursor, WT_VERSION_CUR_HS_EXAUSTED);
+                WT_ERR(WT_NOTFOUND);
+            }
+            break;
+        case WT_PAGE_COL_FIX:
+            /*
+             * If search returned an insert, we might be past the end of page in the append list, so
+             * there's no on-disk value.
+             */
+            if (cbt->recno >= cbt->ref->ref_recno + page->entries) {
+                F_SET(version_cursor, WT_VERSION_CUR_ON_DISK_EXHAUSTED);
+                F_SET(version_cursor, WT_VERSION_CUR_HS_EXAUSTED);
+                WT_ERR(WT_NOTFOUND);
+            }
+            break;
+        case WT_PAGE_COL_VAR:
+            /* Empty page doesn't have any on page value. */
+            if (page->entries == 0) {
+                F_SET(version_cursor, WT_VERSION_CUR_ON_DISK_EXHAUSTED);
+                F_SET(version_cursor, WT_VERSION_CUR_HS_EXAUSTED);
+                WT_ERR(WT_NOTFOUND);
+            }
+            break;
+        default:
+            WT_ERR(__wt_illegal_value(session, page->type));
+        }
+
+        /* Get the ondisk value. */
+        WT_ERR(__wt_value_return_buf(cbt, cbt->ref, &cbt->upd_value->buf, &cbt->upd_value->tw));
+
+        if (!WT_TIME_WINDOW_HAS_STOP(&cbt->upd_value->tw)) {
+            durable_stop_ts = version_cursor->upd_durable_stop_ts;
+            stop_ts = version_cursor->upd_stop_ts;
+            stop_txn = version_cursor->upd_stop_txnid;
+        } else {
+            durable_stop_ts = cbt->upd_value->tw.durable_stop_ts;
+            stop_ts = cbt->upd_value->tw.stop_ts;
+            stop_txn = cbt->upd_value->tw.stop_txn;
+        }
+
+        __wt_cursor_set_value(cursor, cbt->upd_value->tw.start_txn, cbt->upd_value->tw.start_ts,
+          cbt->upd_value->tw.durable_start_ts, stop_txn, stop_ts, durable_stop_ts,
+          WT_UPDATE_STANDARD, cbt->upd_value->tw.prepare, 0, WT_VERSION_DISK_IMAGE);
+
+        upd_found = true;
+        F_SET(version_cursor, WT_VERSION_CUR_ON_DISK_EXHAUSTED);
+    }
+
+    if (!upd_found && !F_ISSET(version_cursor, WT_VERSION_CUR_HS_EXAUSTED)) {
+        /* Ensure we can see all the content in the history store. */
+        F_SET(hs_cursor, WT_CURSTD_HS_READ_COMMITTED);
+
+        if (!F_ISSET(hs_cursor, WT_CURSTD_KEY_INT)) {
+            /* Ensure enough room for a column-store key without checking. */
+            WT_ERR(__wt_scr_alloc(session, WT_INTPACK64_MAXSIZE, &key));
```

Comment:
Shouldn't the size be the max of `WT_INTPACK64_MAXSIZE` and `table_cursor->key.size`?
Probably why you are seeing the crashes, when the table has keys > 9 bytes.